tests/numeric_test.py

```diff
@@ -197,21 +197,398 @@ def testCumProd(self):
         pda = ak.array(na)
+        self.assertTrue(np.allclose(np.cos(na), ak.cos(pda).to_ndarray()))
+
+        na = np.arange(0, 10, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.cos(na), ak.cos(pda).to_ndarray()))
 
+        na = np.linspace(0, 10)
+        pda = ak.array(na)
         self.assertTrue(np.allclose(np.cos(na), ak.cos(pda).to_ndarray()))
+
         with self.assertRaises(TypeError):
             ak.cos([range(0, 10)])
 
+    def testTan(self):
+        na = np.arange(0, 10, dtype="int64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.tan(na), ak.tan(pda).to_ndarray()))
+
+        na = np.arange(0, 10, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.tan(na), ak.tan(pda).to_ndarray()))
+
+        na = np.linspace(0, 10)
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.tan(na), ak.tan(pda).to_ndarray()))
+
+        with self.assertRaises(TypeError):
+            ak.tan([range(0, 10)])
+
+    def testArcsin(self):
+        na = np.arange(-1, 2, dtype="int64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray()))
+
+        na = np.arange(0, 2, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray()))
+
+        na = np.linspace(-1, 1)
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arcsin(na), ak.arcsin(pda).to_ndarray()))
+
+        with self.assertRaises(TypeError):
+            ak.arcsin([range(0, 10)])
+
+    def testArccos(self):
+        na = np.arange(-1, 2, dtype="int64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray()))
+
+        na = np.arange(0, 2, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray()))
+
+        na = np.linspace(-1, 1)
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arccos(na), ak.arccos(pda).to_ndarray()))
+
+        with self.assertRaises(TypeError):
+            ak.arccos([range(0, 10)])
+
+    def testArctan(self):
+        na = np.arange(0, 10, dtype="int64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray()))
+
+        na = np.arange(0, 10, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray()))
+
+        na = np.linspace(0, 10)
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray()))
+
+        with self.assertRaises(TypeError):
+            ak.arctan([range(0, 10)])
+
+        # Edge case: infinities
+        na = np.array([np.inf, -np.inf])
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.arctan(na), ak.arctan(pda).to_ndarray(), equal_nan=True))
+
+    def testArctan2(self):
```

Comment:
This test needs to include testing that the scalar-scalar case properly produces an error message. As it stands, the function logic needs to be modified to produce the expected error and the test needs to be updated to ensure that the case does in fact error.

Reply:
Updated tests